test/Benchmarks/Instrumentation/InstrumentedAspNetCoreBenchmark.cs

```diff
@@ -0,0 +1,78 @@
+﻿// <copyright file="InstrumentedAspNetCoreBenchmark.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System.Net.Http;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+#if NETCOREAPP2_1
+using TestApp.AspNetCore._2._1;
+#else
+using TestApp.AspNetCore._3._1;
+#endif
+
+namespace Benchmarks.Instrumentation
+{
+    [InProcess]
+    [MemoryDiagnoser]
+    public class InstrumentedAspNetCoreBenchmark
+    {
+        private const string SourceName = "http-client-test";
+
+        private HttpClient client;
+        private TracerProvider tracerProvider;
+        private WebApplicationFactory<Startup> factory;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            void ConfigureTestServices(IServiceCollection services)
+            {
+                this.tracerProvider = Sdk.CreateTracerProviderBuilder()
+                    .AddAspNetCoreInstrumentation()
+                    .AddSource(SourceName)
+                    .Build();
+            }
+
+            this.factory = new WebApplicationFactory<Startup>()
+                .WithWebHostBuilder(builder =>
+                    builder.ConfigureTestServices(ConfigureTestServices));
+            this.client = this.factory.CreateClient();
```

Comment:
while this approach works, I am thinking - should we rely on this or instead spinup own Asp.Net Core webserver? I don't know if Testclient/server truly represents real world.

Reply:
something like this:
https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/WEB/Src/DependencyCollector/DependencyCollector.Tests/DependencyTrackingTelemetryModuleTestNetCore.cs#L543

Reply:
just updated to the localServer and updated the values in the description.Let me know what do you think.